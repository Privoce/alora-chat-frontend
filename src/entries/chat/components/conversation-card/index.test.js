import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import {
	ConversationCardComponent,
	ConversationCardSketch
} from 'entries/chat/components';

describe('<ConversationCardComponent />', () => {
	const modelI = {
		profile: {
			label: 'AM',
			width: 40,
			height: 40,
			backgroundColor: 'red',
			color: 'white',
			labelFontSize: 12
		},
		title: {
			text: 'teste',
			fontSize: 14,
			maxWidth: 100,
			margin: '0px 0px 0px 14px'
		},
		desc: {
			text: 'Hey there! how are you',
			fontSize: 13,
			maxWidth: 100,
			margin: '0px 0px 0px 14px'
		}
	};

	const getWrapper = (model) => {
		return shallow((
			<div>
				<ConversationCardComponent {...model} />
				<ConversationCardSketch />
			</div>
		));
	};

	it('renders correctly', () => {
		const wrapper = getWrapper(modelI);
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});
