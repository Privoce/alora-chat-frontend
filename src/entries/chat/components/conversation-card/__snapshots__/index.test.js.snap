// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConversationCardComponent /> renders correctly 1`] = `
<div>
  <ConversationCardComponent
    desc={
      Object {
        "fontSize": 13,
        "margin": "0px 0px 0px 14px",
        "maxWidth": 100,
        "text": "Hey there! how are you",
      }
    }
    profile={
      Object {
        "backgroundColor": "red",
        "color": "white",
        "height": 40,
        "label": "AM",
        "labelFontSize": 12,
        "width": 40,
      }
    }
    title={
      Object {
        "fontSize": 14,
        "margin": "0px 0px 0px 14px",
        "maxWidth": 100,
        "text": "teste",
      }
    }
  />
  <ConversationCardSketch />
</div>
`;
